c++/src/kj/compat/http.c++

```diff
@@ -2073,49 +2073,49 @@ public:
   }
 
   kj::Promise<void> writeBodyData(const void* buffer, size_t size) {
-    KJ_REQUIRE(!writeInProgress, "concurrent write()s not allowed") { return kj::READY_NOW; }
```

Comment:
Interestingly these recovery blocks were leaking memory before (when exceptions are enabled!) due to the longstanding clang bug with throwing post-return.

Or at least, they were leaking memory up until the recent optimization making `kj::READY_NOW` into a singleton.

Reply:
Ouch!